sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -213,22 +211,22 @@ public String getScheme() {
      */
     @Override
     public FileSystem newFileSystem(URI uri, Map<String, ?> config) throws IOException {
-        String accountName = extractAccountName(uri);
+        String endpoint = extractAccountEndpoint(uri);
 
-        if (this.openFileSystems.containsKey(accountName)) {
-            throw LoggingUtility.logError(this.logger, new FileSystemAlreadyExistsException("Name: " + accountName));
+        if (this.openFileSystems.containsKey(endpoint)) {
```

Comment:
Does any additional logic need to change here to prevent two endpoints pointing to the same account but using different styles or containing additional information such as query params?